server/src/main/java/org/elasticsearch/index/get/GetResult.java

```diff
@@ -222,20 +226,34 @@ public String sourceAsString() {
         return sourceAsMap();
     }
 
+
+    public Map<String, DocumentField> getMetadataFields() {
+        return new HashMap<>(metaFields);
+    }
+
+    public Map<String, DocumentField> getDocumentFields() {
+        return new HashMap<>(documentFields);
+    }
+
     public Map<String, DocumentField> getFields() {
+        Map<String, DocumentField> fields = new HashMap<>();
```

Comment:
Do we want to create a new map every time we need to access `getFields` or `field`? May be better to create this `fields` map only once in the constructor?

Reply:
Actually I was asking something similar few patches ago, please see https://github.com/elastic/elasticsearch/pull/38373#discussion_r276022220
got the feeling that it is not the top concern for now, but I don't really have a strong opinion here.

Reply:
@Sandmannn  thanks, I agree with can optimize this in the subsequent PRs.